### PR TITLE
fix cross entropy axis param

### PR DESCRIPTION
**Pull request by awni**
Minor fix to use axis parameter correctly in `losses.cross_entropy_loss`.

### CODE REVIEW

**Comment by awni** [comment]
Also added a speedup for gradient clipping:

Pre: 3.96
Post: 2.57

Benchmark:

```python
import time
import mlx.core as mx
import mlx.optimizers as opt

grads = [mx.random.uniform(shape=(2048, 2048)) for _ in range(10)]

def fn(grads):
    for _ in range(10):
        grads, norm = opt.clip_grad_norm(grads, 1.0)
    return grads

for _ in range(100):
    mx.eval(fn(grads))

tic = time.time()
for _ in range(100):
    mx.eval(fn(grads))
toc = time.time()
print(toc - tic)
```

File: python/mlx/optimizers/optimizers.py [hunk 1]
```diff
@@ -971,10 +971,6 @@ def clip_grad_norm(grads, max_norm):
...
     norm_squared = tree_reduce(lambda acc, g: acc + g.square().sum(), grads, 0.0)
     total_norm = mx.sqrt(norm_squared)
-    normalizer = max_norm / (total_norm + 1e-6)
-
-    def clipper(g):
-        return mx.where(total_norm < max_norm, g, g * normalizer)
-
-    clipped_grads = tree_map(clipper, grads)
+    normalizer = mx.minimum(max_norm / (total_norm + 1e-6), 1.0)
+    clipped_grads = tree_map(lambda g: g * normalizer, grads)
```
Comment (angeloskath) [inline python/mlx/optimizers/optimizers.py:975]: Why do a multiplication when you can do a multiplication and a useless conditional... 🙈

**Review by angeloskath — APPROVED** [review]
👍
